lambda_function.py

```diff
@@ -1,57 +1,55 @@
-    rds_instances = rds_client.describe_db_instances(
-        DBInstanceIdentifier=RDS_IDENTIFIER)
-    rds_endpoint = rds_instances['DBInstances'][0]['Endpoint']['Address']
-    ip_address = socket.gethostbyname(rds_endpoint)
-    rds_port = rds_instances['DBInstances'][0]['Endpoint']['Port']
-
-    # Retrieve the existing target of the target group
-    targets = elbv2_client.describe_target_health(
-        TargetGroupArn=TARGET_GROUP_ARN)
-
-    # Get the current IP address in the target group
-    if targets['TargetHealthDescriptions']:
-        current_ip = targets['TargetHealthDescriptions'][0]['Target']['Id']
-    else:
-        current_ip = None
-
-    # If the IP addresses don't match, update the target group
-    if current_ip and current_ip != ip_address:
-            # Deregister the current target
-            elbv2_client.deregister_targets(
-                TargetGroupArn=TARGET_GROUP_ARN,
-                Targets=[
-                    {
-                        'Id': current_ip
-                    },
-                ]
-            )
+        rds_endpoint = rds_instances['DBInstances'][0]['Endpoint']['Address']
+        ip_address = socket.gethostbyname(rds_endpoint)
+
+        # Retrieve the existing target of the target group
+        target_group_arn = details['target_group_arn']
+        targets = elbv2_client.describe_target_health(TargetGroupArn=target_group_arn)
+
+        # Check and update the target group
+        current_ip = targets['TargetHealthDescriptions'][0]['Target']['Id'] if targets['TargetHealthDescriptions'] else None
+        if current_ip != ip_address:
+            if current_ip:
+                # Deregister the current target
+                elbv2_client.deregister_targets(TargetGroupArn=target_group_arn, Targets=[{'Id': current_ip}])
 
             # Register the new target
-            elbv2_client.register_targets(
-                TargetGroupArn=TARGET_GROUP_ARN,
-                Targets=[
-                    {
-                        'Id': ip_address,
-                        'Port': rds_port
-                    },
-                ]
-            )
+            elbv2_client.register_targets(TargetGroupArn=target_group_arn, Targets=[{'Id': ip_address, 'Port': rds_port}])
+            message = f"Target group {target_group_arn} updated. New target IP: {ip_address}"
+        else:
+            message = f"Target group {target_group_arn} already up to date. Current target IP: {ip_address} and Port: {rds_port}"
+
+        return {'success': True, 'message': message}
+    except Exception as e:
+        return {'success': False, 'message': f"Failed to update targets for {rds_identifier} with error: {e}"}
+
+def lambda_handler(event, context):
+    update_messages = []
+
+    for rds_identifier, details in RDS_DETAILS.items():
+        result = update_target_registration(rds_identifier, details)
+        update_messages.append(result['message'])
 
     return {
         'statusCode': 200,
```

Comment:
do we want to return 200 if there was an exception / failure on a target update?